libcontainer/container_linux.go

```diff
@@ -1249,7 +1254,7 @@ func (c *linuxContainer) orderNamespacePaths(namespaces map[configs.NamespaceTyp
 			if strings.ContainsRune(p, ',') {
 				return nil, newSystemError(fmt.Errorf("invalid path %s", p))
 			}
-			paths = append(paths, p)
+			paths = append(paths, fmt.Sprintf("%s:%s", configs.NsName(nsType), p))
```

Comment:
It's so that we have type safety with `setns`. I've also updated the `setns` code so that it will make sure we're attaching to the right type of namespace. While this isn't a functional change, it does make sure that we don't accidentally join a namespace we didn't intend to.